packages/web-app-files/tests/components/FileUpload.spec.js

```diff
@@ -0,0 +1,64 @@
+import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
+import FileUpload from '../../src/components/FileUpload.vue'
+import DesignSystem from 'owncloud-design-system'
+import stubs from '../../../../tests/unit/stubs'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+
+const Translate = jest.fn()
+
+describe('File Upload Component', () => {
+  const mountOptions = {
+    propsData: {
+      path: ''
+    },
+    localVue,
+    directives: { Translate }
+  }
+
+  it('should render component', () => {
+    const wrapper = mount(FileUpload, mountOptions)
+
+    expect(wrapper.exists()).toBeTruthy()
+  })
+
+  describe('when upload file button is clicked', () => {
+    it('should call "triggerUpload"', async () => {
+      const wrapper = mount(FileUpload, {
+        ...mountOptions,
+        stubs: {
+          ...stubs,
+          'oc-button': false
+        }
+      })
+
+      const spyTriggerUpload = jest.spyOn(wrapper.vm, 'triggerUpload')
+      const uploadButton = wrapper.find('button')
+      const fileUploadInput = wrapper.find('#fileUploadInput')
+      fileUploadInput.element.click = jest.fn()
```

Comment:
you can use mockMethods, so that you don't have to forceupdate the `vm`
see https://github.com/owncloud/web/blob/39b7857cabde66ae96bb11b820a89ac87f5e2f2f/packages/web-runtime/tests/components/Notification.spec.js#L189